### PR TITLE
feat: add image loading placeholder

### PR DESCRIPTION
**Pull request by mdneyazahmad**
### Details

Add a loading indicator while loading an image.
### Fixed Issues

$ https://github.com/Expensify/App/issues/7584

### Tests

1. Open a chat with images
2. Verify that a loading indicator appears for images that are loading
- [x] Verify that no errors appear in the JS console

### QA Steps

1. Same as test steps
- [x] Verify that no errors appear in the JS console

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web

web

#### Mobile Web

mweb

#### Desktop

desktop

#### iOS

ios

#### Android

android

### CODE REVIEW

File: src/components/ImageWithSizeCalculation.js [hunk 1]
```diff
@@ -29,6 +30,14 @@ const defaultProps = {
  * it can be appropriately resized.
  */
 class ImageWithSizeCalculation extends PureComponent {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            loading: true,
+        }
```
Comment (mountiny) [inline src/components/ImageWithSizeCalculation.js:38]: @mdneyazahmad Thank you for the PR! Got one styling mistake here.

**Review by mountiny — COMMENTED** [review]
Bump for lint error, thank you for fixing it. Gonna wait for Rushat for when he gets time to have a look.

File: src/components/ImageWithSizeCalculation.js (hunk 1 above)
Comment (mdneyazahmad) [reply src/components/ImageWithSizeCalculation.js]: Fixed the linting error

Comment (rushatgabhane) [reply src/components/ImageWithSizeCalculation.js]: Thanks for the bump Vit :)

**Comment by rushatgabhane** [comment]
@mdneyazahmad there's a regression - loading indicator isn't show when image is being uploaded. 
Let's fix this 💪

https://user-images.githubusercontent.com/29673073/153815228-9acd7a4c-feac-4771-8930-705fd6271155.mp4

**Comment by mdneyazahmad** [comment]
@rushatgabhane  could you retest? It seems to work as expected.

https://user-images.githubusercontent.com/77761491/153818070-ad81a93a-2ddf-454b-ab64-92a3eb25528f.mp4

**Comment by mdneyazahmad** [comment]
Also after this pr, dimension jumping issue is even more highlighted. That does not look good.

**Comment by rushatgabhane** [comment]
> could you retest? It seems to work as expected.

@mdneyazahmad hmm i can still repro it after hard refresh, and in incognito mode.

**Comment by mdneyazahmad** [comment]
@rushatgabhane  I did the same it still works.

**Comment by mdneyazahmad** [comment]
Could you please do `npm install` and then `npm run web`?

**Comment by rushatgabhane** [comment]
@mdneyazahmad nope that didn't help.

@mountiny could you please help us verify [this](https://github.com/Expensify/App/pull/7723#issuecomment-1038721162) whenever you can, thanks!

**Comment by mountiny** [comment]
Sorry for the delay, I will be able to have a look into this later today or tomorrow the latest.

**Comment by mountiny** [comment]
@mdneyazahmad Hmm, weird, I am getting the same problem as Rushat 😬 
https://user-images.githubusercontent.com/36083550/154113512-1e0c9aa9-3bf3-4884-9716-b5442ffd9d39.mov

**Comment by mdneyazahmad** [comment]
I tested in (Chrome/Firefox/Safari), it works on all these browser. 

> @mdneyazahmad there's a regression - loading indicator isn't show when image is being uploaded.
> Let's fix this 💪

Here is the issue with this feature. https://github.com/Expensify/App/issues/7463

I am not sure why. Could you please try with merging main (if not). If still does not work, I will have to find the root cause for it.

@rushatgabhane

**Comment by mountiny** [comment]
@mdneyazahmad Tested again with merged main in Chrome and the same result. This time the blank placeholder disappeared, then the spinner showed up and then the picture showed up. Lots of jumping, not a great experience. I am not sure what is going on there.

Also I got confused with the linked issue since it looks almost the same. If I understand the difference, this PR should focus on spinner for the image being loaded in chat if not cached, the other one for when it is uploaded.

Either way it seems like it is not working as expected right now, but I am not sure how to help with debugging now, odd you cant reproduce the problem since it happens everytime for me.

**Comment by mountiny** [comment]
I am using local set up which might influence the way/speed the upload is happening but I think Rushat is using the same setup as you do.

**Comment by rushatgabhane** [comment]
Yes, I'm using local setup and it happens everytime.

**Comment by mountiny** [comment]
@mdneyazahmad Any progress on this one? Have you been able to replicate it?

**Comment by okansahin** [comment]
Hello guys,

Sorry I saw this issue recently and no one mentioned me too. I wish you to mention me in here.

First of all, jumping problem is looks bad. But it is already at there before my PR. So there is no relation with the code I sent in PR. It is a different issue. 

As for the problem some of you are having here, it stands before your eyes :)

It's just about @mdneyazahmad uploads by clicking add attachment button, @rushatgabhane and @mountiny upload by dragging. I fixed this and sent to related branch.

Please try to reproduce it with drag&drop before getting branch to confirm me.

Thank you

**Comment by mdneyazahmad** [comment]
Thanks @okansahin for figuring it out. I was able to reproduce it with drag and drop.
@rushatgabhane and @mountiny  Could you confirm that it works when  add attachment button is clicked and uploaded?

File: src/components/ImageWithSizeCalculation.js [hunk 2]
```diff
@@ -29,6 +30,14 @@ const defaultProps = {
  * it can be appropriately resized.
  */
 class ImageWithSizeCalculation extends PureComponent {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            loading: true,
```
Comment (rushatgabhane) [inline src/components/ImageWithSizeCalculation.js:37]: @mdneyazahmad let's rename it to `isLoading`
https://github.com/Expensify/App/blob/9ef244cf828f56d31c033df185f1eda4d2c0db28/STYLE.md#boolean-variables-and-props

Comment (mdneyazahmad) [reply src/components/ImageWithSizeCalculation.js]: @rushatgabhane done

**Review by rushatgabhane — CHANGES_REQUESTED** [review]
@mdneyazahmad for the jumping issue.
We could reduce the number of jumps from 2 to 1, if we use the same thumbnail styles as the upload preview thumbnail.

What do you think? Can we do any better?

https://user-images.githubusercontent.com/29673073/155143949-bb178c6d-4c48-491c-8674-407e92691b37.mp4

**Comment by mountiny** [comment]
@rushatgabhane how many jumps are there without this PR? If there is just one and this PR changes it to 2, then we should fix it here.

But if there already were 2 jumps before this PR, we should keep that in a separate job just to be atomic with the issues. Feel free to report the issue/bug!

**Comment by mountiny** [comment]
@okansahin Aaand thank you very much for this of course! Really appreciate your help 🙇

**Comment by rushatgabhane** [comment]
There are 2 jumps without this PR.

> we should keep that in a separate job just to be atomic with the issues.

Sure! That makes sense, thanks for jumping in :)

**Review by rushatgabhane — APPROVED** [review]
@mountiny LGTM! 🎉 Tests well on all platforms.

Kudos to @okansahin for connecting the dots 🙌
I've commented on #7463 to get the upload preview fixed

**Comment by okansahin** [comment]
Thanks to you too, all. 🙌

**Review by mountiny — APPROVED** [review]
Nice! Thank you @mdneyazahmad for the patience and work on this PR. thanks to @okansahin for help as well and great job @rushatgabhane testing and reviewing it!

**Comment by roryabraham** [comment]
This PR was confirmed to cause a [regression](https://github.com/Expensify/App/issues/7900), and has been reverted.

**Comment by Stutikuls** [comment]
**Issue 1 -**

**Title**- [Medium]: Chrome+ Jaws : Screen reader : Name & Role is not defined for 'Actions (+)' control.
**Actual** - Screen reader is reading some other information on Actions(+) control.
**Expected** - Role = 'Button' and Name = 'Actions/Add attachments' should defined for Actions(+) control.
**WCAG failure** - 4.1.2
**Reproducible in staging?**: - Yes
**Version Number** : - v1.1.40-2
**Platforms** - Web (Chrome + Jaws), iOS, Desktop (MAC)

**Note** - Due to installation of build not able to check on Android

https://user-images.githubusercontent.com/97600950/157381005-478db150-6c68-4017-8b83-08a120dc5785.mp4

**Issue 2 -**

**Title**- [Medium]: Chrome+ Jaws : Keyboard : 'Actions (+)' control is not accessible using screen reader.
**Actual** -  'Actions (+)' control is not accessible using screen reader, while press Enter on the control focus moves to another control.
**Expected** - Actions (+)' control should accessible using screen reader.
**WCAG failure** - 2.1.1
**Reproducible in staging?:** - Yes
**Version Number:** - v1.1.40-2
**Platforms** - Web (Chrome + Jaws), iOS, Mobile-web(iOS)

Note - Due to installation of build not able to check on Android

https://user-images.githubusercontent.com/97600950/157382341-ae72e99d-b121-444e-bd35-56c23846fae4.mp4

**Issue 3 -**

**Title**- [Medium]: Chrome+ Jaws : Screen reader: Role is not defined for 'Send' control.
**Actual** -  'Actions (+)' control is not accessible using screen reader, while press Enter on the control focus moves to another control.
          **Note-** On MAC screen reader is reading "Send Group".
**Expected** - Actions (+)' control should accessible using screen reader.
**WCAG failure** - 4.1.2
**Reproducible in staging?:** - Yes
**Version Number:** - v1.1.40-2
**Platforms** - Web (Chrome + Jaws), iOS, Mobile-web (iOS), Desktop(MAC)

Note - Due to installation of build not able to check on Android


https://user-images.githubusercontent.com/97600950/157382967-689c127b-42b6-4e12-974c-3256cb9c4010.mp4

**Issue 4 -**

**Title**- [Medium]: Chrome+ Jaws : Screen reader: Status message is not being announced by screen reader after activate the 'Send' control.
**Actual** -  Status message is not being announced by screen reader after activate the 'Send' control. Screen reader is silent and focus lands on the Action(+) button.
**Expected** - Status message should announce by screen reader after uploading the attachment. Screen reader should read like "Image/video has been uploaded".
**WCAG failure** - 4.1.3
**Reproducible in staging?:** - Yes
**Version Number:** - v1.1.40-2
**Platforms** - Web (Chrome + Jaws), iOS, Mobile-web (iOS), Desktop(MAC)

Note - Due to installation of build not able to check on Android


https://user-images.githubusercontent.com/97600950/157387589-2e93ffe0-bd3e-4fac-85ed-be96c3832b81.mp4

**Pull request merged**